lib/widgets/search/main_search_bar.dart

```diff
@@ -0,0 +1,129 @@
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: <Widget>[
+      Container(
+        margin: new EdgeInsets.only(left: 15.0, top: MediaQuery.of(context).padding.top + 5.0, right: 15.0),
+        padding: EdgeInsets.only(bottom: 10),
+        child: Container(
+          decoration: BoxDecoration(
+              borderRadius: BorderRadius.circular(20), color: Colors.white),
+          child: Row(
+            children: <Widget>[
+              Container(
+                child: IconButton(
+                        splashColor: Colors.grey,
+                        icon: Icon(Icons.menu),
+                        onPressed: () => Scaffold.of(context).openDrawer(),
+                      ),
+              ),
+              Expanded(
+              child: TextField(
+                readOnly: true,
+                cursorColor: Colors.black,
+                keyboardType: TextInputType.text,
+                textInputAction: TextInputAction.go,
+                decoration: InputDecoration(
+                    border: InputBorder.none,
+                    contentPadding: EdgeInsets.symmetric(horizontal: 15),
+                    hintText: "Search"),
+                onTap: (){
+                  showSearch(
+                      context: context,
+                      delegate: PositionedFloatingSearchBar());
+                },
+              ),
+            ),
+              Padding(
+                padding: const EdgeInsets.only(right: 8.0),
+                child: RaisedButton(
+                child: Text(campus),
+                textColor: Colors.white,
+                color: Color.fromRGBO(147, 35, 57, 1),
+                shape: RoundedRectangleBorder(
+                    borderRadius: new BorderRadius.circular(50)),
+                onPressed: () {
+                  setState(() {
+                    (campus == 'SGW') ? (campus = 'LOY') : (campus = 'SGW');
+                  });
+                },
+              ),
+              )],
+          ),
+        ),
+      ),
+    ]);
+  }
+}
+
+class PositionedFloatingSearchBar extends SearchDelegate<String> {
+
+final Function(LatLng) latlng;
+PositionedFloatingSearchBar({this.latlng});
+
+  @override
+  List<Widget> buildActions(BuildContext context) {
+    return [
+      IconButton(
+        icon: Icon(Icons.clear),
+        onPressed: () {
+          query = "";
+        },
+      ),
+    ];
+  }
+
+  @override
+  Widget buildLeading(BuildContext context) {
+
+    return IconButton(
+        icon: AnimatedIcon(
+          icon: AnimatedIcons.menu_arrow,
+          progress: transitionAnimation,
+        ),
+        onPressed: () {
+          close(context, null);
+        });
+  }
+
+  @override
+  Widget buildSuggestions(BuildContext context) {
+    final suggestionList = query.isNotEmpty
+        ? []
+        : []
+            .where((p) => p.getRoomNumber().startsWith(query.toUpperCase()))
```

Comment:
This looks off. Getter for RoomCoordinate room number is roomId

Reply:
Modified this to match current models.